pkg/distributor/distributor_test.go

```diff
@@ -4836,6 +4834,11 @@ func TestHandleIngesterPushError(t *testing.T) {
 			ingesterPushError:         createStatusWithDetails(t, codes.Unavailable, testErrorMsg, mimirpb.INVALID).Err(),
 			expectedIngesterPushError: newIngesterPushError(createStatusWithDetails(t, codes.Unavailable, testErrorMsg, mimirpb.INVALID)),
 		},
+		"a DedlineExceeded gRPC ingester error an ingesterPushError with INVALID cause": {
+			// This is how context.DeadlineExceeded error is translated into a gRPC error.
+			ingesterPushError:         status.Error(codes.DeadlineExceeded, context.DeadlineExceeded.Error()),
+			expectedIngesterPushError: newIngesterPushError(createStatusWithDetails(t, codes.Unavailable, context.DeadlineExceeded.Error(), mimirpb.INVALID)),
```

Comment:
This change seems fine to me but do we want to consider adding a new status or renaming `INVALID`? `INVALID` makes it seem like this should never happen.

Reply:
I agree with this @56quarters. The main purpose of `INVALID` when it was introduced was to have a default cause value that would be assigned to the error details if somebody "forgets" to assign it explicitly. Do you have any suggestion? WDYT about `UNKNOWN`?

Reply:
That works for me!